src/Expensify.js

```diff
@@ -109,6 +118,11 @@ class Expensify extends PureComponent {
                     PushNotification.clearNotifications();
                 }
 
+                // Force offline mode on staging or dev if needed
+                if (_.contains([CONST.ENVIRONMENT.STAGING, CONST.ENVIRONMENT.DEV], this.props.environment)) {
+                    Network.setIsOffline(this.props.network.shouldForceOffline);
+                }
```

Comment:
`constructor()` will run before the component mounts because wrapping the component in `withOnyx()` delays rendering the child (i.e. this component is the `WrappedComponent` in the linked code below) until all values are read from storage - therefore whatever value is in storage (if they exist) will be accessible to the `constructor()`. 

https://github.com/Expensify/react-native-onyx/blob/91aabb7219ddc8722cc2955de0398b1ae70c546d/lib/withOnyx.js#L160-L181

Reply:
Ah thank you that's very helpful!